File: Algorithms/p409_Longest_Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        array<int,256> cnt = {};
        for(char c : s){
            ++cnt[c];
        }
        int res = 0;
        int has_odd = 0;
        for(int num : cnt){
            res += num/2*2;
            has_odd |= num&0x1;
        }
        return res+has_odd;
    }
};

File: Algorithms/p409_Longest_Palindrome/p409_Longest_Palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        cnt = [0]*256
        for c in s:
            cnt[ord(c)] += 1
        res = 0
        has_odd = 0
        for num in cnt:
            res += num//2*2
            has_odd |= num&0x1
        return res+has_odd
